packages/server/lib/plugins/child/run_require_async_child.js

```diff
@@ -78,12 +80,35 @@ function run (ipc, configFile, projectRoot) {
     return false
   }
 
-  ipc.on('loadConfig', () => {
+  // Config file loading of modules is tested within
+  // system-tests/projects/config-cjs-and-esm/*
+  const loadConfig = async (configFile) => {
+    // We try to use `bundleRequire` to handle user-space node imports where possible.
+    // The one exception is that when
+    // - users have a JS config file *and*
+    // - a tsconfig in their project *and*
+    // - that tsconfig has a `target` or es5/es2015/es3
+    const tsConfig = loadTsConfig(process.cwd())?.data
+    const shouldUseNodeImport = tsConfig && !configFile.endsWith('.ts') && tsConfig.compilerOptions?.target?.match(/es(2015|5|3)/ig)
+
+    if (shouldUseNodeImport) {
+      debug(`We're loading the configFile via user's node instead of bundleRequire. TSConfig target is`, tsConfig.compilerOptions?.target, `and the config file ends with .ts`)
+
+      return await import(configFile)
+    }
+
+    debug(`We're loading the configFile via bundleRequire`)
+
+    return (await bundleRequire({ filepath: configFile })).mod
+  }
+
+  ipc.on('loadConfig', async () => {
     try {
       debug('try loading', configFile)
-      const exp = require(configFile)
+      const configFileExport = await loadConfig(configFile)
```

Comment:
Have we tried the try/catch approach & using this as a fallback? Based on the test failures it looks like this would be a pretty big breaking change if we went with this as the default path if we can even get it to work

Reply:
Yeah, I started with the try/catch approach. It was buggy. `esbuild-register` didn't support typescript files with type: module because it requires certain node runtime flags.

I don't think the test failures are related to this, I think they're related to the errors changes.

Gimme a bit to debug. It works extremely well in the 12+ system test projects I have so I'm doubtful that this is actually unstable.